### ISSUE TITLE
No data movement to GPU when data from desc with type keyword

### ISSUE THREAD

**Issue by QingleiCao** [ICLDisco/parsec#643]
## Describe the bug

In PTG, when data is from memory and a `type` keyword is provided, there will be no data movement to GPU, e.g.,

`READ A <- descA(m, n)  [ type = FULL ]`

The reason is the data movement to GPU will be skipped here: https://github.com/ICLDisco/parsec/blob/4f76b6d2fa622b5e7ba96cd09a3333415acba220/parsec/mca/device/device_gpu.c#L1323C1-L1336C69.

### To Reproduce
1. Checkout version 'commit-hash'
2. Compile with the following options '....'  [e.g., `head config.log`]
3. Run test '....' [e.g. `ctest --output-on-failure -R xyz`]

## Environment (please complete the following information):
 - PaRSEC version: [e.g., git hash]
 - OS: [e.g. CentOS/7]
 - Compiler: [e.g. GCC/7.3.0]
 - MPI version: [e.g. Open MPI/4.1.2]

## Additional context
The content of the `config.log` file can be useful in some cases.

**Comment by bosilca** [comment]
I don't understand why we skipped version 0 data transfer. Does it work if you remove that line ?

**Comment by QingleiCao** [comment]
> I don't understand why we skipped version 0 data transfer. Does it work if you remove that line ?

I'm not sure the reason; maybe because of reshape of desc? 

Yeah, it works if removing that line.

**Comment by abouteiller** [comment]
The test is for a pure NEW data, which is not the case here. So the question is why the data-in is NULL for a read from a collection?

**Comment by bosilca** [comment]
That's what the comment states, not what the check does. In the case @QingleiCao describes there is no predecessor, so source_repo_entry is indeed NULL, so the case is similar to `NEW`. What I don't understand is why the dc of the data_in is NULL if the read is from a valid collection ?

**Comment by QingleiCao** [comment]
This desc will be reshaped to arena_ADT. Is it related to here?

**Comment by bosilca** [comment]
yes, that now makes sense. If the original data is reshaped, then the reshaed data does not belong to a DC, so it iwll take the same path as the NEW data. Moveover, as the version is 0 (because it is the first of such type), it will completely fit the check.

We need to figure out a different way to check for the NEW data, one that does not rely on empirical checks but on stricter rules.

**Comment by abouteiller** [comment]
Related to #564

**Comment by bosilca** [comment]
should be now fixed.
